nat-lab/tests/utils/batching.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+import math
+import os
+import pytest
+import subprocess
+import typing
+from scapy.all import PcapReader, Packet  # type: ignore
+from typing import Callable, List, Optional
+
+
+def _generate_histogram(
+    data: list[int], buckets: int, bucket_size: int = 1
```

Comment:
Maybe float is more relevant if you consider batch sizes smaller than 1?

Reply:
Having sub 1second would require floats but that would also require more work to convert seconds to milliseconds so it would work, thus I think it's unjustified at the moment, basically don't have it because we don't need it yet.